src/XMakeBuildEngine/UnitTests/BackEnd/TargetUpToDateChecker_Tests.cs

```diff
@@ -417,8 +417,13 @@ public void InputItemsTransformedToDifferentNumberOfOutputsTwoWays()
         public void MultiInputItemsThatCorrelatesWithMultipleTransformOutputItems2()
         {
             Console.WriteLine("MultiInputItemsThatCorrelatesWithMultipleTransformOutputItems2");
-            MockLogger logger = new MockLogger();
-            Project p = new Project(XmlReader.Create(new StringReader(ObjectModelHelpers.CleanupFileContents(@"
+            string currentDirectory = Directory.GetCurrentDirectory();
+
+            try
+            {
+                Directory.SetCurrentDirectory(ObjectModelHelpers.TempProjectDir);
```

Comment:
The files created by the inline project are created in the project directory, which, since it's inline, is the current directory. Set current directory to temp so the files are created there. The garbage in the source tree after a test run is driving me nuts. I fixed two such issues. Hopefully it's a start.